students/eowyn/session04/trigrams.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+
+import random
+import string
+
+# pg 125 of think python has line processing example
+PUNCTUATION = ['.', '!', '?']
+
+# TO DO: figure out how to properly capitalize I, proper nouns
+# TO DO: Fix trigrams that end in words like "and", prepositional phrases, articles
+
+
+def convert_text(filename):
+    ''' 
+    Convert the full text of a local filename (str) to 
+    return the corresponding dictionary of trigrams
+    '''
+    words = []
+    for lines in open(filename):
+        words += process_line(lines)
+    trigrams = {}  # a dictionary
+    for i in range(len(words) - 2):
+        # doesn't include the last char, so this is a pair of words
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        trigrams.setdefault(pair,[]).append(follower)
+    return trigrams
+
+
+def construct_text(filename, n):
+    ''' Given a filename (str) and number, n, construct a 
+    made-up text of n random trigrams from the text contained in filename.
+    n should be an int and will otherwise be rounded to int.
+    '''
+    n = round(n)
+    trigrams = convert_text(filename)  # dictionary
+    new_words = []
+    for triplet in range(n):
+        # randomly select a LIST key from the dictionary
+        randokey = random.choice(list(trigrams))
+        # randomly select a STRING value from the pre-selected key
+        randoval = random.choice(trigrams[randokey])
+        new_words += [randokey[0]] + [randokey[1]] + \
+            [randoval]  # append triplet onto new word list
+    process_output(new_words)
+    new_words = " ".join(new_words)
+    print(new_words)
+
+
+def process_line(line):
+    '''
+    Remove punctionation, white space, and upper case from lines
+    of text and return list of remaining words
+    '''
+
+    line.replace("-", " ")
+    line.replace("--", " ")
+    words = []
+    for word in line.split():
+        word = word.strip(string.punctuation + string.whitespace)
+        word = word.lower()
+        words += [word]
+    return words
+
+
+def process_output(word_list):
+    '''
+    Add random punctuation followed by capitalization to a list of words. Sentences are 
+    between 4 and 18 words long hard-coded but must be multiples of 3 to have them 
+    begin/end where trigrams begin/end.
+    '''
+    i = 0
+    while i < len(word_list):
```

Comment:
could you do this with a `for i in range(something)` ?

maybe `range(random.randint(...)` or ??